book_source/05_developer_guide/03_git/01_using-git.Rmd

```diff
@@ -29,229 +11,230 @@ easy to do.
 
 The forked repository is the only way for external people to commit code back to PEcAn and BETY. The pull request will start a review process that will eventually result in the code being merged into the main copy of the codebase. See https://help.github.com/articles/fork-a-repo for more information,  especially on how to keep your fork up to date with respect to the original. (Rstudio users should also see [Git + Rstudio](Using-Git.md#git--rstudio), below)
 
 You can setup SSH keys to make it easier to commit cod back to GitHub. This might especially be true if you are working from a cluster, see [set up ssh keys](https://help.github.com/articles/generating-ssh-keys)
 
 1. Introduce yourself to GIT
 
-        git config --global user.name "FULLNAME"
-        git config --global user.email you@yourdomain.example.com
+`git config --global user.name "FULLNAME"`
+`git config --global user.email you@yourdomain.example.com`
+
+2. Fork PEcAn on GitHub. Go to the PEcAn source code and click on the Fork button in the upper right. This will create a copy of PEcAn in your personal space.
 
-2. Fork PEcAn on GitHub. Goto the PEcAn source code and click on the Fork button in the upper right. This will create a copy of PEcAn in your personal space.
+3. Clone to your local machine via command line
 
-3. Clone to your local machine 
+`git clone git@github.com:<username>/pecan.git`
 
-        git clone git@github.com:<username>/pecan.git
+If this does not work, try the https method
 
-3. Define upstream repository
- 
-        cd pecan
-        git remote add upstream git@github.com:PecanProject/pecan.git
+`git clone https://github.com/PecanProject/pecan.git`
 
+4. Define upstream repository
+
+```
+cd pecan
+git remote add upstream git@github.com:PecanProject/pecan.git
+```
 
 ### During development:
 
 * commit often; 
-* each commit can address 0 or 1 issue; many commits can reference an issue (see [[Link commits to issue|Using-Git#link-commits-to-issues]])
-* ensure that all tests are passing before anything is pushed into master.
-
-#### Basic Workflow
+* each commit can address 0 or 1 issue; many commits can reference an issue
+* ensure that all tests are passing before anything is pushed into develop.
 
-PLEASE DO NOT USE THIS, SEE ADVANCED WORKFLOW!
+### Basic Workflow
 
+We do not recommend using this workflow is you plan on continuously contributing to PEcAn
 1. Get the latest code from the main repository
 
-        git pull upstream master
+`git pull upstream develop`
 
 2. Do some coding
 
 3. Commit after each chunk of code (multiple times a day)
-        
-        git commit -m "<some descriptive information about what was done; references/fixes gh-X>"
+
+`git commit -m "<some descriptive information about what was done; references/fixes gh-X>"`
 
 4. Push to YOUR Github (when a feature is working, a set of bugs are fixed, or you need to share progress with others)
 
-        git push origin <branchname>
+`git push origin develop`
```

Comment:
Would it make sense to encourage pushing to named feature branches even in personal forks? I realize this is the generic workflow and not PEcAn-specific, but it's good practice everywhere -- otherwise it's forever "username/develop and upstream/develop  have diverged"

Reply:
master and develop are special branches that normally should never be worked in.

Reply:
This is the "Basic Workflow" and serves as an easy stepping stone to learning/contributing. That's why I have the "We do not recommend using this workflow if you plan on continuously contributing to PEcAn" warning in this section. That being said, I can make it more clear why branching and the "Recommended Workflow" on line 134 should be used over this section.